### PR TITLE
Update datasource plugin template

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:
Updates the frontend part of the datasource templates to use functional components and stop using legacy form fields.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@0.11.0-canary.172.d1fcc2b.0
  npm install @grafana/sign-plugin@0.2.0-canary.172.d1fcc2b.0
  # or 
  yarn add @grafana/create-plugin@0.11.0-canary.172.d1fcc2b.0
  yarn add @grafana/sign-plugin@0.2.0-canary.172.d1fcc2b.0
  ```
</details>

### CODE REVIEW

File: packages/create-plugin/templates/datasource/src/components/ConfigEditor.tsx [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-const { SecretFormField, FormField } = LegacyForms;
-
 interface Props extends DataSourcePluginOptionsEditorProps<MyDataSourceOptions> {}
 
-interface State {}
-
-export class ConfigEditor extends PureComponent<Props, State> {
-  onPathChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onOptionsChange, options } = this.props;
+export const ConfigEditor = (props: Props) => {
```
Comment (mckn) [inline packages/create-plugin/templates/datasource/src/components/ConfigEditor.tsx:8]:
```suggestion
export function ConfigEditor(props: Props): ReactElement {
```

File: packages/create-plugin/templates/datasource/src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -1,47 +1,32 @@
...
 import { MyDataSourceOptions, MyQuery } from '../types';
 
-const { FormField } = LegacyForms;
-
 type Props = QueryEditorProps<DataSource, MyQuery, MyDataSourceOptions>;
 
-export class QueryEditor extends PureComponent<Props> {
-  onQueryTextChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onChange, query } = this.props;
+export const QueryEditor = ({ query, onChange, onRunQuery }: Props) => {
```
Comment (mckn) [inline packages/create-plugin/templates/datasource/src/components/QueryEditor.tsx:9]:
```suggestion
export function QueryEditor({ query, onChange, onRunQuery }: Props): ReactElement {
```

File: packages/create-plugin/templates/backend/pkg/plugin/datasource.go [hunk 3]
```diff
@@ -73,10 +74,12 @@ func (d *Datasource) query(_ context.Context, pCtx backend.PluginContext, query
 
 	err := json.Unmarshal(query.JSON, &qm)
 	if err != nil {
-		return backend.ErrDataResponse(backend.StatusBadRequest, "json unmarshal: " + err.Error())
+		return backend.ErrDataResponse(backend.StatusBadRequest, fmt.Sprintf("json unmarshal: %v", err))
```
Comment (mckn) [inline packages/create-plugin/templates/backend/pkg/plugin/datasource.go:77]:
```suggestion
		return backend.ErrDataResponse(backend.StatusBadRequest, fmt.Sprintf("json unmarshal: %v", err.Error()))
```

**Review by mckn — APPROVED** [review]
Added some small nits but looks good!

**Comment by jackw** [comment]
@iwysiu thanks so much for contributing to create-plugin. 🏅 👏

**Pull request merged**